internal/nodes/avi_model_evh_nodes.go

```diff
@@ -1020,3 +1134,281 @@ func (o *AviObjectGraph) BuildPolicyRedirectForVSForEvh(vsNode []*AviEvhVsNode,
 	}
 
 }
+
+// RouteIngrDeletePoolsByHostname : Based on DeletePoolsByHostname, delete pools and policies that are no longer required
+func RouteIngrDeletePoolsByHostnameForEvh(routeIgrObj RouteIngressModel, namespace, objname, key string, fullsync bool, sharedQueue *utils.WorkerQueue) {
+	ok, hostMap := routeIgrObj.GetSvcLister().IngressMappings(namespace).GetRouteIngToHost(objname)
+	if !ok {
+		utils.AviLog.Warnf("key: %s, msg: nothing to delete for route: %s", key, objname)
+		return
+	}
+
+	utils.AviLog.Debugf("key: %s, msg: hosts to delete are :%s", key, utils.Stringify(hostMap))
+	for host, hostData := range hostMap {
+		shardVsName := DeriveHostNameShardVSForEvh(host, key)
+		if hostData.SecurePolicy == lib.PolicyPass {
+			shardVsName = lib.GetPassthroughShardVSName(host, key)
+		}
+		if shardVsName == "" {
+			// If we aren't able to derive the ShardVS name, we should return
+			utils.AviLog.Infof("key: %s, shard vs ndoe not found for host: %s", host)
+			return
+		}
+		modelName := lib.GetModelName(lib.GetTenant(), shardVsName)
+		found, aviModel := objects.SharedAviGraphLister().Get(modelName)
+		if !found || aviModel == nil {
+			utils.AviLog.Warnf("key: %s, msg: model not found during delete: %s", key, modelName)
+			continue
+		}
+
+		// Delete the pool corresponding to this host
+		if hostData.SecurePolicy == lib.PolicyEdgeTerm {
+			aviModel.(*AviObjectGraph).DeletePoolForHostnameForEvh(shardVsName, host, routeIgrObj, hostData.PathSvc, key, true, true, true)
+		} else if hostData.SecurePolicy == lib.PolicyPass {
+			aviModel.(*AviObjectGraph).DeleteObjectsForPassthroughHost(shardVsName, host, routeIgrObj, hostData.PathSvc, key, true, true, true)
+		}
+		if hostData.InsecurePolicy == lib.PolicyAllow {
+			aviModel.(*AviObjectGraph).DeletePoolForHostnameForEvh(shardVsName, host, routeIgrObj, hostData.PathSvc, key, true, true, false)
+		}
+		ok := saveAviModel(modelName, aviModel.(*AviObjectGraph), key)
+		if ok && len(aviModel.(*AviObjectGraph).GetOrderedNodes()) != 0 && !fullsync {
+			PublishKeyToRestLayer(modelName, key, sharedQueue)
+		}
+	}
+	// Now remove the secret relationship
+	routeIgrObj.GetSvcLister().IngressMappings(namespace).RemoveIngressSecretMappings(objname)
+	utils.AviLog.Infof("key: %s, removed ingress mapping for: %s", key, objname)
+
+	// Remove the hosts mapping for this ingress
+	routeIgrObj.GetSvcLister().IngressMappings(namespace).DeleteIngToHostMapping(objname)
+
+	// remove hostpath mappings
+	updateHostPathCacheV2(namespace, objname, hostMap, nil)
+}
+
+// CRD functions
+
+func BuildL7HostRuleForEvh(host, namespace, ingName, key string, vsNode *AviEvhVsNode) {
```

Comment:
Looks like this method is an exact replica of the HostRule method in crd_translator.go file's BuildL7HostRule - can we keep them same by defining an interface and supply EVH VS node to one and the regular VS node to the other? The EVH VS and the regular VS would have to implement a common interface for this. cc @monotosh-avi

Reply:
+1 I had sample code of interfacing out the evh and normal vs struct. Did not proceed as there were many checks in main code which I did not want to add. In future if the other CRD code will be deprecated then this can be replaced later. We can discuss this over the meeting.

Reply:
Added the interface to handle this.